syntax = "proto3";

option java_outer_classname = "PredictionService";
package nightTemp;
option java_multiple_files = true;
option java_package = "com.rain.farm.app.report.grpc";

service TempService {
    rpc GetTemp (TempRequest) returns (TempResponse) {}
}

message TempRequest {
    repeated TempPrediction tempPrediction = 1;
}

message TempPrediction {
    string cityName = 1;
    string weatherText = 2;
    string weatherCode = 3;
    string temperature = 4;
    string humidity = 5;
    string windDirection = 6;
    string windSpeed = 7;
    int64 timestamp = 8;
}

message TempResponse {
    repeated TempPrediction tempPrediction = 1;
}
